source/IJSVGGradientLayer.m

```diff
@@ -34,9 +36,171 @@ - (void)drawInContext:(CGContextRef)ctx
+    }
+    
+    __block CGPoint startPoint;
+    __block CGPoint endPoint;
+    
+    void (^applyTransform)(CGAffineTransform transform) = ^(CGAffineTransform transform) {
+        startPoint = CGPointApplyAffineTransform(startPoint, transform);
+        endPoint = CGPointApplyAffineTransform(endPoint, transform);
+    };
+    
+    if([self.gradient isKindOfClass:[IJSVGLinearGradient class]]) {
+        
+        // basic start and end
+        startPoint = (CGPoint) {
+            .x = self.gradient.x1.value,
+            .y = self.gradient.y1.value
+        };
+        
+        endPoint = (CGPoint) {
+            .x = self.gradient.x2.value,
+            .y = self.gradient.y2.value
+        };
+        
+        // only do these calculations if the type is %
+        if(self.gradient.x2.type == IJSVGUnitLengthTypePercentage) {
+            
+            // move origin back due to user space
+            if(isUserSpace == YES) {
+                absolutePosition.x = (absolutePosition.x/CGRectGetWidth(boundingBoxRect));
+                absolutePosition.y = (absolutePosition.y/CGRectGetHeight(boundingBoxRect));
+                
+                // apply the absolute transform
+                CGAffineTransform transform;
+                transform = CGAffineTransformMakeTranslation(-absolutePosition.x,
+                                                             -absolutePosition.y);
+                applyTransform(transform);
+            }
+            
+            // apply the transforms specified on the gradient element
+            for(IJSVGTransform * gradientTransform in self.gradient.transforms) {
+                applyTransform(gradientTransform.CGAffineTransform);
+            }
+            
+            // conver the points
+            if(isUserSpace == YES) {
+                
+                // start
+                startPoint.x = ((startPoint.x - CGRectGetMinX(objectRect))/CGRectGetWidth(objectRect));
+                startPoint.y = ((startPoint.y - CGRectGetMinY(objectRect))/CGRectGetHeight(objectRect));
+                
+                // end
+                endPoint.x = ((endPoint.x - CGRectGetMaxX(objectRect))/CGRectGetWidth(objectRect)) + 1.f;
+                endPoint.y = ((endPoint.y - CGRectGetMaxY(objectRect))/CGRectGetHeight(objectRect)) + 1.f;
+            }
+            
+            // convert back to pixel values
+            startPoint.x = CGRectGetWidth(objectRect)*startPoint.x;
+            startPoint.y = CGRectGetHeight(objectRect)*startPoint.y;
+            
+            endPoint.x = CGRectGetWidth(objectRect)*endPoint.x;
+            endPoint.y = CGRectGetHeight(objectRect)*endPoint.y;
+            
+        } else {
+            
+            // apply the transforms specified on the gradient element
+            for(IJSVGTransform * gradientTransform in self.gradient.transforms) {
+                applyTransform(gradientTransform.CGAffineTransform);
+            }
```

Comment:
Most of the code is from the merge, I added this transform to adhere to the spec for SVG.